### PR TITLE
Cluster Serving NCF Example

### PR DESCRIPTION
**Pull request by respecteverything**

### CODE REVIEW

File: apps/cluster-serving-ncf/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+* Python 3.6/3.7
+* JDK 8
+* Apache Spark 2.x (This version needs to be same with the version you use to build Analytics Zoo)
+* Redis 5.0 and above
+
+## Install or download Analytics Zoo  
+Follow the instructions [here](https://analytics-zoo.github.io/master/#PythonUserGuide/install/) to install analytics-zoo via __pip__ or __download the prebuilt package__.  
+
+## Prepare the model
+!Should we add our model online?
```
Comment (Litchilitchy) [inline apps/cluster-serving-ncf/README.md:15]: just prepare the link

File: apps/cluster-serving-ncf/README.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+You can download ncf-bigdl model using this [link](....)
+
+## Run after pip install 
+* Step one: 
+
+You can simply run commend : `cluster-serving init`. It will set the envs for you.
```
Comment (Litchilitchy) [inline apps/cluster-serving-ncf/README.md:21]: typo: run command

File: apps/cluster-serving-ncf/README.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  # must be provided given data_type is tensor. eg: [1,2] (tensor) [[1],[2,1,2],[3]] (table)
+  tensor_shape: [[1],[1]]
+  # default, topN(1)
+  filter:
+...
+``` 
+
+* Step three: 
+
+Run `cluster-serving start`. Then you will see the lod in your terminal.
```
Comment (Litchilitchy) [inline apps/cluster-serving-ncf/README.md:45]: log

File: apps/cluster-serving-ncf/README.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+* Step four: 
+
+Turn on another terminal. Run `Python input.py` then you can see the output `Result is :[[2,0.72023946]]`
```
Comment (Litchilitchy) [inline apps/cluster-serving-ncf/README.md:48]: Open another terminal

Comment (Litchilitchy) [inline apps/cluster-serving-ncf/README.md:48]: `python input.py`

**Comment by jason-dai** [comment]
Move to pyzoo/zoo/examples/serving/

File: apps/cluster-serving-ncf/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+```
+export ANALYTICS_ZOO_HOME=/path/to/your/prebuild/package
+source $ANALYTICS_ZOO_HOME/bin/analytics-zoo-env.sh
+cd $ANALYTICS_ZOO_HOME/bin/cluster-serving
+```
+More information about cluster serving installation, please check the link [here.](https://analytics-zoo.github.io/master/#ClusterServingGuide/ProgrammingGuide/)
+
+* Step two: 
+
+After step one, you can repeat step two & step three to load your model with cluster serving.
```
Comment (qiyuangong) [inline apps/cluster-serving-ncf/README.md:64]: `repeat` is not good. `Follow` ?

File: apps/cluster-serving-ncf/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+* Step three: 
+
+Turn on another terminal. Before run python program, you need to reset the env args:
```
Comment (qiyuangong) [inline apps/cluster-serving-ncf/README.md:68]: Turn on another terminal, and reset the env args:

File: pyzoo/zoo/examples/serving/Recommendation-ncf/input.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+from zoo.serving.client import InputQueue, OutputQueue
+import numpy as np
+# input tensor
+x = np.array([2, 3], dtype=np.float32)
+
+input_api = InputQueue()
+input_api.enqueue_tensor('my_input', x)
+
+output_api = OutputQueue()
+img1_result = output_api.query('my_input')
```
Comment (jason-dai) [inline pyzoo/zoo/examples/serving/Recommendation-ncf/input.py:10]: why img1_result?

Comment (Litchilitchy) [reply pyzoo/zoo/examples/serving/Recommendation-ncf/input.py:10]: ……will change to result, and together with new schema update in next PR

**Pull request merged**